synfig-core/src/modules/mod_geometry/advanced_outline.cpp

```diff
@@ -203,7 +203,14 @@ namespace {
 			iterator i1 = lower_bound(p);
 			if (i1 == end())
 				{ clear(); return; }
-				
+
+			if (approximate_equal(i1->first, p)) {
+				erase(begin(), i1);
+				i1->second.side0 = side;
+				i1->second.pp0 = Vector(p, 0);
```

Comment:
You should change the iterator 'contents' before possibly invalidating it with `erase()`. So, move the `erase` code line to below the `i1->second.xxxx` code lines.